Lib/http/server.py

```diff
@@ -562,6 +563,10 @@ def log_error(self, format, *args):
 
         self.log_message(format, *args)
 
+    # https://en.wikipedia.org/wiki/List_of_Unicode_characters#Control_codes
+    __control_char_table = str.maketrans(
+            {c: '!' for c in itertools.chain(range(0x20), range(0x7f,0xa0))})
```

Comment:
This transformation removes information. Would it be possible to escape control characters instead? For example, encode them as "\xHH" (4 characters, code point as hexadecimal) or "%HH" (3 characters, in hexadecimal).

Reply:
done. though realize that someone cannot reliably reconstruct the actual raw value from the log message as a request with \x in it will come through as is and cannot be differentiated from our \xHH for a control character. I think this is fine in a bugfix.

Changing the log_record `"%s"` to `%r` in 3.12 would be nice but would be more disruptive to do in a bugfix as it'd change the quoting and do things like escape `\`s.  Perhaps make that change in 3.12?